src/ripple/app/misc/impl/LoadFeeTrack.cpp

```diff
@@ -81,56 +82,110 @@ LoadFeeTrack::lowerLocalFee ()
 
 //------------------------------------------------------------------------------
 
-// NIKB TODO: Once we get C++17, we can replace lowestTerms
-//            with this:
-//
-// template <class T1, class T2,
-//     class = std::enable_if_t<
-//         std::is_integral_v<T1> &&
-//         std::is_integral_v<T2>>
-// >
-// void lowestTerms(T1& a,  T2& b)
-// {
-//     if (auto const gcd = std::gcd(a, b))
-//     {
-//         a /= gcd;
-//         b /= gcd;
-//     }
-// }
+// EHENNIS TODO: Once we get C++17, we can replace gcd with std::gcd
+
+template <class T>
+std::enable_if_t<std::is_unsigned_v<T>, T>
+abs(T t)
+{
+    return t;
+}
+
+template <class T>
+std::enable_if_t<std::is_signed_v<T>, T>
+abs(T t)
+{
+    return std::abs(t);
+}
 
 template <class T1, class T2,
     class = std::enable_if_t <
-        std::is_integral<T1>::value &&
-        std::is_unsigned<T1>::value &&
+        std::is_integral_v<T1> &&
         sizeof(T1) <= sizeof(std::uint64_t) >,
     class = std::enable_if_t <
-        std::is_integral<T2>::value &&
-        std::is_unsigned<T2>::value &&
+        std::is_integral_v<T2> &&
         sizeof(T2) <= sizeof(std::uint64_t) >
 >
-void lowestTerms(T1& a,  T2& b)
+std::common_type_t<T1, T2> gcd(T1 const a, T2 const b)
 {
     if (a == 0 && b == 0)
-        return;
+        return 0;
 
-    std::uint64_t x = a, y = b;
+    std::uint64_t x = abs(a), y = abs(b);
     while (y != 0)
     {
         auto t = x % y;
         x = y;
         y = t;
     }
-    a /= x;
-    b /= x;
+    return x;
 }
 
+namespace detail
+{
+
+struct xrp_unit_product_tag;
+
+using xrp_unit_product =
+    units::TaggedFee<detail::xrp_unit_product_tag, std::uint64_t>;
```

Comment:
I don't think I'd define this type. I see why it's done, tho. What do you think about defining a function `mulDiv(FeeUnit64, XRPAmount, FeeUnit64) -> XRPAmount` and don't use a tagged type inside that function?

Reply:
Duh. The templated unit-handling version of that function in feeunits.h already does this. So that's an easy update, and then I can remove `xrp_unit_product_tag` and functions.